### ISSUE TITLE
Offer an alternative interface for specifying registy/repo/tag triple with docker-style syntax

### ISSUE THREAD

**Issue by mishak87** [mayflower/docker-ls#5]
When listing tags of official repository results in error.

``` sh
> docker-ls tags redis
ERROR: authorization rejected by registry
```

Listing it with `library/` prefix works.

``` sh
> docker-ls tags library/redis
requesting list . done
repository: library/redis
tags:
- "2"
...
```

Would it make sense to prepend `library/` for images without `<vendor>/`?

**Comment by DirtyHairy** [comment]
Hi @mishak87!

Afaik, the `<vendor>/<name>` scheme is just a convention, and there is nothing that keeps you for creating a repo that does not have a vendor prefix (I would have to test though). Anyway, I'd rather not add any magic for resolving repository names, but I am very much inclined to add a note concerning the official registry to the documentaton.

Cheers
-Christian

**Comment by mishak87** [comment]
Yes v2 [does not limit `<name>`](https://github.com/docker/distribution/blob/master/docs/spec/api.md#overview).
This use-case is only when using public docker registry (aka Hub; which is default).
I think it would be more user friendly to prefix them in this case - especially for new docker users.

**Comment by DirtyHairy** [comment]
Documented in 2621ced .

**Comment by mishak87** [comment]
OT: Just a thought. Using [docker/reference.WithName](https://github.com/docker/docker/blob/master/reference/reference.go#L75-L88) could simplify CLI input so registry does not have to be specified via argument. By simulating docker behaviour however it is messy would fit more into docker toolset (docker, docker-compose image names).

``` sh
> docker-ls tags super.secret/vendor/project
```

**Comment by DirtyHairy** [comment]
Hi @mishak87 !

I'll think about it. While I think that the current interface is more clear and concise, it might be worthwhile to add a mode of operation where you specify the resource the docker way. Of course, I am also open to pull requests :smirk:

Cheers
-Christian
